### PR TITLE
Print player

### PR DESCRIPTION
**Pull request by t-h2o**
# Player

## Move

Position X, Y
Angle [0 - (2 * PI)]

D: Turn right
A: Turn left
W: Go front
S: Go back
```
 W
ASD 
```

### CODE REVIEW

File: headers/cub3d.h [hunk 1]
```diff
@@ -29,14 +29,26 @@
...
 
 // Path for textures
 # define MM_GROUND	"assets/textures/minimap/ground.xpm"
 # define MM_WALL	"assets/textures/minimap/wall.xpm"
+# define MM_PLAYER	"assets/textures/minimap/player.xpm"
 
 # define MAPSDIR	"assets/maps/"
 
+// Math
+# define PI			3.14159
```
Comment (Giggzzz) [inline headers/cub3d.h]: is PI defined in math.h ?

File: srcs/mini_map.c [hunk 2]
```diff
@@ -16,31 +16,76 @@
...
 // Print the mini map
+// Print the player
+// Print the data
 void	print_minimap(t_info *info)
 {
 	int		pos_x;
 	int		pos_y;
 
+	mlx_clear_window
+		(((t_info *)info)->mlx[INIT], ((t_info *)info)->mlx[WINDOW]);
```
Comment (Giggzzz) [inline srcs/mini_map.c]:
```suggestion
		(info->mlx[INIT], info->mlx[WINDOW]);
```

File: srcs/window.c [hunk 3]
```diff
@@ -18,6 +18,14 @@ static int
...
 	if (key == KEY_ESC)
 		close_game(info);
+	if (key == KEY_W || key == KEY_UP)
+		player_move(info, SENSE_MOVE);
+	if (key == KEY_S || key == KEY_DOWN)
+		player_move(info, -SENSE_MOVE);
+	if (key == KEY_D || key == KEY_RIGHT)
+		player_angle(info, SENSE_ROTATION);
+	if (key == KEY_A || key == KEY_LEFT)
+		player_angle(info, -SENSE_ROTATION);
```
Comment (Giggzzz) [inline srcs/window.c]:
```suggestion
	else if (key == KEY_W || key == KEY_UP)
		player_move(info, SENSE_MOVE);
	else if (key == KEY_S || key == KEY_DOWN)
		player_move(info, -SENSE_MOVE);
	else if (key == KEY_D || key == KEY_RIGHT)
		player_angle(info, SENSE_ROTATION);
	else if (key == KEY_A || key == KEY_LEFT)
		player_angle(info, -SENSE_ROTATION);
```

File: srcs/window.c [hunk 4]
```diff
@@ -30,7 +38,7 @@ int	start_window(t_info *info)
 		(info->mlx[INIT], W_WIDTH, W_HEIGHT, "Cub3D");
 	if (!info->mlx[WINDOW])
 		return (error_msg("Couldn't create window"));
-	mlx_key_hook(info->mlx[WINDOW], deal_key, info);
+	mlx_hook(info->mlx[WINDOW], 2, 1, deal_key, info);
```
Comment (Giggzzz) [inline srcs/window.c:41]:
```suggestion
	mlx_hook(info->mlx[WINDOW], 2, 1L << 0, deal_key, info);
```
does it work on linux/mac/wsl without that change?

File: headers/cub3d.h [hunk 5]
```diff
@@ -59,15 +72,30 @@ enum e_mlx {
...
+// position X and Y
+// A is the angle
+// Player delta X
+// Player delta Y
+enum e_pp {
+	X,
+	Y,
+	A,
+	PDX,
+	PDY
```
Comment (Giggzzz) [inline headers/cub3d.h]:
```suggestion
enum e_pp {
	X,
	Y,
	A,
	DX,
	DY
```

File: srcs/player.c [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+{
+	info->pos_player[X] += distance * info->pos_player[PDX];
+	info->pos_player[Y] += distance * info->pos_player[PDY];
+	print_minimap(info);
+}
+
+// When the player change his rotation,
+// That update the delta X and Y when the player move
+void
+	player_angle(t_info *info, float rotation)
```
Comment (Giggzzz) [inline srcs/player.c]:
```suggestion
	player_rotate(t_info *info, float rotation)
```

File: headers/cub3d.h (hunk 1 above)
Comment (t-h2o) [reply headers/cub3d.h]: in `man 3 math`

```
MATHEMATICAL CONSTANTS
     In addition to the functions listed above,
     math.h defines a number of useful constants, listed below.

     CONSTANT        VALUE
     M_E             base of natural logarithm, e
     M_LOG2E         log2(e)
     M_LOG10E        log10(e)
     M_LN2           ln(2)
     M_LN10          ln(10)
     M_PI            pi
     M_PI_2          pi / 2
     M_PI_4          pi / 4
     M_1_PI          1 / pi
     M_2_PI          2 / pi
     M_2_SQRTPI      2 / sqrt(pi)
     M_SQRT2         sqrt(2)
     M_SQRT1_2       sqrt(1/2)
```
I will change PI with M_PI

Comment (t-h2o) [reply headers/cub3d.h]: Done: 7284dabb950ec34e4b7f417b2a3539bee3da7bc2

File: headers/cub3d.h (hunk 5 above)
Comment (t-h2o) [reply headers/cub3d.h]: Done: bb7bb64b0b2c95ec6d15c3b62884c2edeb4e3228

File: srcs/player.c (hunk 6 above)
Comment (t-h2o) [reply srcs/player.c]: Done: fc67e3bd6a14a00131cd4e8c567e0e612d0abe77

File: srcs/window.c (hunk 3 above)
Comment (t-h2o) [reply srcs/window.c]: Done: aaa9517086b73d95ad0596d6468565c5c6895b55

**Review by Giggzzz — APPROVED** [review]
good work! lgtm

File: srcs/window.c (hunk 4 above)
Comment (t-h2o) [reply srcs/window.c:41]: That work on my GNU/Linux

**Pull request merged**